### DISCUSSION TITLE
BTLE iOS not seen when phone is locked

### DISCUSSION THREAD

**Discussion by ghost** [mKeRix/room-assistant#698] [General]
Hi,
First off - incredible work with room-assistant.

I wanted to check in to make sure it wasn’t a configuration or device issue on my end. When utilising the iOS app - room-assistant is unable to detect device nearby when iPhone is locked.

I assume this is one of those “why can’t Apply just allow people to choose how their device functions” kind of situations but wanted to double check.

Obviously having RA detect BTLE configured iOS devices while locked in your pocket for room based actions “upon entry” would be awesome.

Cheers

**Comment by mKeRix** [comment]
The iOS app should be detectable in the background as well - the Bluetooth permissions are requested for background advertising as well, which should work when the screen is off. If the Bluetooth stack crashes on your phone for whatever reason room-assistant isn't able to restart advertising until the app is opened again though (OS limitations).

I'm not sure if some iPhone models behave differently - I tested it with a bunch of different devices, but can't cover everything obviously. Official documentation for this stuff is kinda sparse to be honest.

(sorry for the late reply!)

**Reply by gasecki** [reply]
I noticed iOS 15.2, device is not detected after reboot till the time of first unlock, after that the app runs in backgroung
